cmd/proxy/proxy.go

```diff
@@ -124,39 +128,49 @@ func runCmd(staticConfiguration *static.Configuration) error {
 	return nil
 }
 
-func setupServer(staticConfiguration *static.Configuration) (*server.Server, error) {
-	providerAggregator := aggregator.NewProviderAggregator(*staticConfiguration.Providers)
+func setupServer(maeshStaticConfiguration *static.Configuration) (*server.Server, error) {
+	traefikStaticConfiguration := maeshStaticConfiguration.ToTraefikConfig()
```

Comment:
The `ToTraefikConfig()` and having two variables could be solved by embedding the traefik configuration in the proxy configuration directly. You'd then use `proxyCfg.Configuration` to access the traefik config, wdyt?

Reply:
The providers are different

Reply:
and all of the middlewarebuilders etc in this function take the traefik static config as arguments, so it is easier at this point to output a compatable struct

Reply:
fixed.